crates/bevy_pbr/src/light_probe/generate.rs

```diff
@@ -0,0 +1,1084 @@
+        irradiance_constants_buffer.write_buffer(&render_device, &queue);
+
+        // create a 2d array view
+        let irradiance_map =
+            env_map_light
+                .diffuse_map
+                .texture
+                .create_view(&TextureViewDescriptor {
+                    dimension: Some(TextureViewDimension::D2Array),
+                    ..Default::default()
+                });
+
+        let irradiance_bind_group = render_device.create_bind_group(
+            "irradiance_bind_group",
+            &layouts.irradiance,
+            &BindGroupEntries::with_indices((
+                (0, &textures.environment_map.default_view),
+                (1, &samplers.linear),
+                (2, &irradiance_map),
+                (3, &irradiance_constants_buffer),
+                (4, &stbn_texture.texture_view),
+            )),
+        );
+
+        // Create copy bind group (source env map → destination mip0)
+        let src_view = env_map_light
+            .environment_map
+            .texture
+            .create_view(&TextureViewDescriptor {
+                dimension: Some(TextureViewDimension::D2Array),
+                ..Default::default()
+            });
+
+        let dst_view = create_storage_view(&textures.environment_map.texture, 0, &render_device);
+
+        let copy_bind_group = render_device.create_bind_group(
+            "copy_mip0_bind_group",
+            &layouts.copy,
+            &BindGroupEntries::with_indices(((0, &src_view), (1, &dst_view))),
+        );
+
+        commands.entity(entity).insert(GeneratorBindGroups {
+            spd: spd_bind_group,
+            radiance: radiance_bind_groups,
+            irradiance: irradiance_bind_group,
+            copy: copy_bind_group,
+        });
+    }
+}
+
+/// Helper function to create a storage texture view for a specific mip level
+fn create_storage_view(texture: &Texture, mip: u32, _render_device: &RenderDevice) -> TextureView {
+    texture.create_view(&TextureViewDescriptor {
+        label: Some(format!("storage_view_mip_{mip}").as_str()),
+        format: Some(texture.format()),
+        dimension: Some(TextureViewDimension::D2Array),
+        aspect: TextureAspect::All,
+        base_mip_level: mip,
+        mip_level_count: Some(1),
+        base_array_layer: 0,
+        array_layer_count: Some(texture.depth_or_array_layers()),
+        usage: Some(TextureUsages::STORAGE_BINDING),
+    })
+}
+
+/// SPD Node implementation that handles both parts of the downsampling (mips 0-12)
+pub struct SpdNode {
+    query: QueryState<(
+        Entity,
+        Read<GeneratorBindGroups>,
+        Read<RenderEnvironmentMap>,
+    )>,
+}
+
+impl FromWorld for SpdNode {
+    fn from_world(world: &mut World) -> Self {
+        Self {
+            query: QueryState::new(world),
+        }
+    }
+}
+
+impl Node for SpdNode {
+    fn update(&mut self, world: &mut World) {
+        self.query.update_archetypes(world);
+    }
+
+    fn run(
```

Comment:
New rendering stuff should add spans https://github.com/bevyengine/bevy/pull/19191.

Reply:
resolved by https://github.com/mate-h/bevy/pull/8